language/move-stdlib/src/natives/debug.rs

```diff
@@ -25,34 +27,73 @@ pub struct PrintGasParameters {
     pub base_cost: InternalGas,
 }
 
+#[allow(dead_code)]
+fn is_string_type(context: &NativeContext, ty: &Type, move_std_addr: AccountAddress) -> bool {
+    let ty_tag = match context.type_to_type_tag(ty) {
+        Ok(ty_tag) => ty_tag,
+        Err(_) => return false,
+    };
+
+    match ty_tag {
+        TypeTag::Struct(struct_tag) => {
+            struct_tag.address == move_std_addr
+                && struct_tag.module.as_str().eq("string")
+                && struct_tag.name.as_str().eq("String")
+        }
+        _ => false,
+    }
+}
+
 #[inline]
 fn native_print(
     gas_params: &PrintGasParameters,
     _context: &mut NativeContext,
     mut ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
+    move_std_addr: AccountAddress,
 ) -> PartialVMResult<NativeResult> {
     debug_assert!(ty_args.len() == 1);
     debug_assert!(args.len() == 1);
 
     // No-op if the feature flag is not present.
     #[cfg(feature = "testing")]
     {
-        let _ty = ty_args.pop().unwrap();
-        let r = pop_arg!(args, Reference);
+        let ty = ty_args.pop().unwrap();
+
+        match is_string_type(_context, &ty, move_std_addr) {
+            true => {
+                let r = pop_arg!(args, StructRef);
+
+                let field_ref = r
+                    .borrow_field(0)? // borrows the 'bytes' field of the std::string::String struct
+                    .value_as::<Reference>()?;
+
+                let bytes = field_ref.read_ref()?.value_as::<Vec<u8>>()?;
```

Comment:
I think you can make a `&[u8]` without need to make a copy, checkout `string.rs` which always gets away without copying.

Reply:
Ah, will take a look.

Reply:
Fixed!